src/content-types/activity.ts

```diff
@@ -1,16 +1,16 @@
 import BaseContentType from './base-content';
-import UIData from './ui-data';
 
 // An activity is a way for players to spend time in between each month of gameplay.
 // e.g. leisure, studying, working
 class Activity extends BaseContentType {
 	// Run any state changing effects of the activity.
-	apply: (/* player */) => void;
+	apply: (/* player */) => void = () => {
+		console.log('warning: apply not set on activity with id ' + this.id);
+		return;
+	};
 
-	constructor(id: string, uiData: UIData, applyFunction: () => void) {
-		super(id, uiData);
-		this.apply = applyFunction;
-	}
+	// for deserialization purposes
+	scriptPath: string = '';
```

Comment:
This is what I'm thinking of to attach scripts. There are some pros and cons to this approach, but it lets us take advantage of TypeScript.